numpy/distutils/mingw32ccompiler.py

```diff
@@ -64,8 +64,8 @@ def __init__ (self,
         # we need to support 3.2 which doesn't match the standard
         # get_versions methods regex
         if self.gcc_version is None:
-            p = subprocess.Popen(['gcc', '-dumpversion'], shell=True,
-                                 stdout=subprocess.PIPE)
+            p = subprocess.Popen(
+                    ['gcc', '-dumpversion'], stdout=subprocess.PIPE)
             out_string = p.stdout.read()
             p.stdout.close()
```

Comment:
```suggestion
            try:
                out_string  = subprocess.check_output(['gcc', '-dumpversion'])
            except OSError, CalledProcessError:
                out_string = ""  # ignore failures to match old behavior
```

Reply:
A bit more cleaning up, but looks nice to me, thanks.